synedrion/src/paillier/keys.rs

```diff
@@ -356,4 +409,34 @@ mod tests {
         let sk = SecretKeyPaillier::<PaillierTest>::random(&mut OsRng).to_precomputed();
         let _pk = sk.public_key();
     }
+
+    #[test]
+    fn debug_redacts_secrets() {
+        let sk = SecretKeyPaillier::<PaillierTest>::random(&mut OsRng);
+
+        let debug_output = format!("Sikrit {:?}", sk);
+        assert_eq!(debug_output, "Sikrit [REDACTED synedrion::paillier::keys::SecretKeyPaillier<synedrion::paillier::params::PaillierTest>]");
+    }
+
+    #[test]
+    fn serialization_and_clone_works() {
+        let mut rng = rand_chacha::ChaCha8Rng::seed_from_u64(123456);
+        let sk = SecretKeyPaillier::<PaillierTest>::random(&mut rng);
+
+        // Can't compare SecretKeyPaillier so we serialize and compare output tokens in this awkward way.
```

Comment:
Actually, should we just implement `Eq` for it? Should be safe, right?

Reply:
Yeah, it's probably fine. It requires `expose_secret()` and I wanted to minimize the number of places we do that, but it's probably ok to have `PartialEq` impl'd.